crates/ruma-identifiers-validation/src/room_version_id.rs

```diff
@@ -8,7 +8,7 @@ pub fn validate(s: &str) -> Result<(), Error> {
         Err(Error::Empty)
     } else if s.chars().count() > MAX_CODE_POINTS {
         Err(Error::MaximumLengthExceeded)
-    } else if !s.chars().all(|c| c.is_alphanumeric() || ".-".contains(c)) {
+    } else if !s.chars().all(|c| c.is_ascii_alphanumeric() || ".-".contains(c)) {
```

Comment:
Probably more efficient alternative:

```suggestion
    } else if !s.bytes().all(|b| b.is_ascii_alphanumeric() || b == b'.' || b == b'-') {
```